tests/integration/live_tests.py

```diff
@@ -0,0 +1,112 @@
+        if bandwidth > norm:
+            actual = bandwidth / norm - 1.0
+            self.assertLess(actual, margin, self._err_message(margin, actual))
+        elif bandwidth < norm:
+            actual = norm / bandwidth - 1.0
+            self.assertLess(actual, margin, self._err_message(margin, actual))
+        else:
+            self.assertEqual(norm, bandwidth)
+
+    def _do_test(self, cases, tcp_free_rate=None, udp_free_rate=None):
+        if not tcp_free_rate:
+            tcp_free_rate = self.tcp_free_rate
+        assert tcp_free_rate or udp_free_rate
+        argv_list = ['tc', '-i', 'lo', '-c']
+        for case in cases:
+            argv_list.extend(('--dclass', case))
+
+        if VERBOSITY:
+            argv_list.append('-v')
+        live_test = LtcLiveTargetRun(argv_list, udp_sendrate='10mbit', duration=DURATION)
+        live_test.run()
+        for case in cases:
+            assert ('tcp' in case and tcp_free_rate) or ('udp' in case and udp_free_rate)
+            free_rate = tcp_free_rate if 'tcp' in case else udp_free_rate
+            branch = parse_branch(case)
+            results = live_test.result[case]
+            self._check_bandwidth(convert2bps(branch['rate']), results['left_in'], MAX_SHAPED_TOLERANCE)
+            self._check_bandwidth(free_rate, results['left_out'], MAX_FREE_RATE_TOLERANCE)
+            self._check_bandwidth(convert2bps(branch['rate']), results['right_in'], MAX_SHAPED_TOLERANCE)
+            self._check_bandwidth(free_rate, results['right_out'], MAX_FREE_RATE_TOLERANCE)
+
+    def test_simple_tcp(self):
+        print('test_simple_tcp BEGIN')
+        cases = ['tcp:9100-9110:512kbit']
+        self._do_test(cases)#, tcp_free_rate=4500000000)
+        print('test_simple_tcp END')
+
+    def test_double_tcp(self):
+        print('test_double_tcp BEGIN')
+        cases = ['tcp:9200:1mbit', 'tcp:9300-9310:512kbit']
+        self._do_test(cases)#, tcp_free_rate=4500000000)
+        print('test_double_tcp END')
+
+    def test_simple_udp(self):
+        print('test_simple_udp BEGIN')
+        cases = ['udp:9400-9410:2mbit']
+        self._do_test(cases, udp_free_rate=10000000)
+        print('test_simple_udp END')
+
+    def test_double_udp(self):
+        print('test_double_udp BEGIN')
+        cases = ['udp:9500:4mbit', 'udp:9600-9610:2mbit']
+        self._do_test(cases, udp_free_rate=10000000)
+        print('test_double_udp END')
+
+    def test_ingress_simple(self):
+        print('test_ingress_simple BEGIN')
+        cases = ['tcp:9700-9710:1mbit']
+        self._do_test(cases)#, tcp_free_rate=6000000000)
+        print('test_ingress_simple END')
+
+    def test_ingress_complex(self):
+        print('test_ingress_complex BEGIN')
+        cases = ['tcp:9800:786kbit', 'udp:9900-9910:3mbit']
+        self._do_test(cases, udp_free_rate=10000000)#, tcp_free_rate=6000000000, udp_free_rate=10000000)
```

Comment:
How about removing the comments here and on line 77, 83, 101 if we don't need them